Make vicuna finetuning serving self-contained (#1855)
skypilot-org/skypilot: examples/llm/vicuna/README.md

pip install git+https://github.com/skypilot-org/skypilot.git
sky check
```

> Find the Vicuna [training SkyPilot YAMLs](train.yaml) and [serving SkyPilot YAMLs](serve.yaml).

Reviewer: ```suggestion
See the Vicuna SkyPilot YAMLs: for [training](train.yaml) and for [serving](serve.yaml).
```

## Serve the official Vicuna model by yourself with SkyPilot

1. Start the serving the Vicuna-7B model on a single A100 GPU:
```bash
sky launch -c vicuna-serve -s serve.yaml
```

> 2. Check the output of the command. There will be a sharable gradio link (like the last line in of the following). Open it in your browser to chat with Vicuna.

Reviewer: ```suggestion
2. Check the output of the command. There will be a sharable gradio link (like the last line of the following). Open it in your browser to chat with Vicuna.
```

```

## Training Vicuna with SkyPilot
Currently, training requires GPUs with 80GB memory.  See `sky show-gpus --all` for supported GPUs.

We can start the training of Vicuna model on the dummy data [dummy.json](dummy.json)[^1] **with a single command**. It will automatically find the available cheapest VM on any cloud.

**To train on your own data**, replace the file with your own, or change the line `/data/mydata.json: ./dummy.json` to the path of your own data in the [train.yaml](train.yaml).

> [^1]: The dummy data was original from the official Vicuna repository, [FastChat](https://github.com/lm-sys/FastChat).

Reviewer: ```suggestion
[^1]: The dummy data was originally from the official Vicuna repository, [FastChat](https://github.com/lm-sys/FastChat).
```
